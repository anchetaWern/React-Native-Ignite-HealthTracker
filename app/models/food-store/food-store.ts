import { Instance, SnapshotIn, SnapshotOut, types } from "mobx-state-tree"

import { FoodModel, FoodSnapshotIn } from "../food/food"
/**
 * Model description here for TypeScript hints.
 */

export const FoodStoreModel = types
  .model("FoodStore")
  .props({
    foods: types.optional(types.array(FoodModel), []),
  })
  .views((self) => ({
    get allFoods() {
      return self.foods
    },
  })) // eslint-disable-line @typescript-eslint/no-unused-vars
  .actions((self) => ({
    saveFood: (foodSnapshot: FoodSnapshotIn) => {
      self.foods.push(foodSnapshot)
    },
  })) // eslint-disable-line @typescript-eslint/no-unused-vars

export interface FoodStore extends Instance<typeof FoodStoreModel> {}
export interface FoodStoreSnapshotOut extends SnapshotOut<typeof FoodStoreModel> {}
export interface FoodStoreSnapshotIn extends SnapshotIn<typeof FoodStoreModel> {}
export const createFoodStoreDefaultModel = () => types.optional(FoodStoreModel, {})
