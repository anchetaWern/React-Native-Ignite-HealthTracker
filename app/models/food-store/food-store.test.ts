import { FoodStoreModel } from "./food-store"

test("can be created", () => {
  const instance = FoodStoreModel.create()

  instance.saveFood({
    id: "somerandomid123",
    name: "fried chicken",
    rating: 2,
  })

  expect(instance.allFoods).toStrictEqual([
    {
      id: "somerandomid123",
      name: "fried chicken",
      rating: 2,
    },
  ])
})
