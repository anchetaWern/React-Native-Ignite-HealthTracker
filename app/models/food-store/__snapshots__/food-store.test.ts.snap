// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be created 1`] = `
Object {
  "foods": Array [
    Object {
      "id": "somerandomid123",
      "name": "fried chicken",
      "rating": 2,
    },
  ],
}
`;
